x/incentives/types/params.go

```diff
@@ -0,0 +1,60 @@
+package types
+
+import (
+	"fmt"
+
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+// Parameter store keys
+var (
+	KeyBlocksPerEpoch = []byte("BlocksPerEpoch")
+)
+
+// ParamTable for minting module.
+func ParamKeyTable() paramtypes.KeyTable {
+	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+func NewParams(blocksPerEpoch int64) Params {
+	return Params{
+		BlocksPerEpoch: blocksPerEpoch,
+	}
+}
+
+// default minting module parameters
+func DefaultParams() Params {
+	return Params{
+		BlocksPerEpoch: 10,
+	}
+}
+
+// validate params
+func (p Params) Validate() error {
+	if err := validateBlocksPerEpoch(p.BlocksPerEpoch); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+// Implements params.ParamSet
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyBlocksPerEpoch, &p.BlocksPerEpoch, validateBlocksPerEpoch),
+	}
+}
+
+func validateBlocksPerEpoch(i interface{}) error {
```

Comment:
What is this doing?

Reply:
It's what others do for params verification. same for all the other on-chain param verifiers.

Reply:
Oh I see, so this is ran in ChangeParameterProposals on the new parameter